x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/IlmHealthIndicatorService.java

```diff
@@ -173,6 +201,11 @@ public class IlmHealthIndicatorService implements HealthIndicatorService {
     public IlmHealthIndicatorService(ClusterService clusterService, StagnatingIndicesFinder stagnatingIndicesFinder) {
         this.clusterService = clusterService;
         this.stagnatingIndicesFinder = stagnatingIndicesFinder;
+        var clusterSettings = this.clusterService.getClusterSettings();
+
+        clusterSettings.addSettingsUpdateConsumer(MAX_TIME_ON_ACTION_SETTING, stagnatingIndicesFinder::recreateRules);
+        clusterSettings.addSettingsUpdateConsumer(MAX_TIME_ON_STEP_SETTING, stagnatingIndicesFinder::recreateRules);
+        clusterSettings.addSettingsUpdateConsumer(MAX_RETRIES_PER_STEP_SETTING, stagnatingIndicesFinder::recreateRules);
```

Comment:
This feels a bit strange to me. I'm wondering if we're mixing responsibilities here a bit?

Like, the constructor for `IlmHealthIndicatorService` subscribes to some settings in order for the `StagnatingIndicesFinder` to update some internal state. 

Should we have the rules configured within `StagnatingIndicesFinder`, and the `StagnatingIndicesFinder` subscribes to cluster settings changes and takes care of these rules? 
This way the `StagnatingIndicesFinder` encapsulates everything related to creating, updating, and evaluating the rules, and the `IlmHealthIndicatorService` will be just a consumer/client.

What do you think?

Reply:
you're totally right about that. I thought the same yesterday, but, later, by mistake, removed the commit changing it along with other small changes. Will rewrite this

Reply:
changed